Libraries/src/Amazon.Lambda.RuntimeSupport/Bootstrap/LambdaBootstrap.cs

```diff
@@ -137,10 +151,41 @@ private LambdaBootstrap(HttpClient httpClient, LambdaBootstrapHandler handler, L
             // and then shut down cleanly. Useful for profiling or running local tests with the .NET Lambda Test Tool. This environment
             // variable should never be set when function is deployed to Lambda.
             var runOnce = string.Equals(Environment.GetEnvironmentVariable(Constants.ENVIRONMENT_VARIABLE_AWS_LAMBDA_DOTNET_DEBUG_RUN_ONCE), "true", StringComparison.OrdinalIgnoreCase);
+            
+            
+            if (_initializer != null && !(await InitializeAsync()))
+            {
+                return;
+            }
+#if NET8_0_OR_GREATER
+            // Check if Initialization type is SnapStart, and invoke the snapshot restore logic.
+            if (_configuration.IsInitTypeSnapstart)
+            {
+                InternalLogger.GetDefaultLogger().LogInformation($"In LambdaBootstrap, Initializing with SnapStart.");
 
-            bool doStartInvokeLoop = _initializer == null || await InitializeAsync();
+                object registry = null;
+                try
+                {
+                    registry = SnapstartHelperCopySnapshotCallbacksIsolated.CopySnapshotCallbacks();
+                }
+                catch (TypeLoadException ex)
+                {
+                    Client.ConsoleLogger.FormattedWriteLine(
```

Comment:
This doesn't appear to terminate or exit the call which means it will fall through where registry will be null. The following comment says `no expection in calling SnapStart hooks` but it looks like there could have been an exception still. Is this correct?